wagtail/contrib/search_promotions/views.py

```diff
@@ -167,7 +189,10 @@ def delete(request, query_id):
     query = get_object_or_404(Query, id=query_id)
 
     if request.method == 'POST':
-        query.editors_picks.all().delete()
+        editors_picks = query.editors_picks.all()
+        for search_pick in editors_picks:
+            log(search_pick, 'wagtail.delete')
```

Comment:
question: could `log()` take on a queryset or a list to do the heavy lifting? or is this done to be explicit in the places that we need it to log in bulk?

Reply:
That could be a future enhancement, yep. I don't know if we'd gain much from it, though - I think the places like this where the bulk action code _doesn't_ already have a loop to put the `log` line in are likely to be pretty rare.